### PR TITLE
First steps towards a dark theme

### PR DESCRIPTION
**Pull request by niik**
This PR adds the first piece of what will become a dark theme for GitHub Desktop. #4780 and #4810 preceded this PR in order to get us to a point where we could start working on a dark theme. The final colors and contrasts have not been locked in yet, think of this as a preview of roughly what we're thinking.

screen shot 2018-06-01 at 15 30 30
screen shot 2018-06-01 at 15 29 56

### Enabling the dark theme

This PR only adds the style sheet itself and doesn't contain any method to select the dark theme in the UI. For testing purposes the theme can be enabled by setting the `theme` localStorage variable to `dark` **and reloading the app**.

We'll work on a proper UI for selecting a theme as well as being able to switch themes without reloading the app in a follow-up PR.

```js
// To enable the dark theme, open DevTools and execute
localStorage.setItem('theme', 'dark');
```
```js
// To revert to default theme
localStorage.removeItem('theme')
```

### Illustration woes

We've had some excellent help from @OiYouYeahYou (#3135, #3773) and @nickcraver (https://github.com/desktop/desktop/issues/1515#issuecomment-388525591) towards making our illustrations dynamically styleable :heart:

Unfortunately that work hasn't made it into this PR. There are unfortunate edge cases with both approaches that we haven't had time to address. In the monochrome svgs colorized with css filters approach the `feColorMatrix` approach didn't want cooperate and always rendered the colors slightly off, I imagine this is due to use of different color spaces or perhaps simply incompetence on my part but I couldn't afford to spend more time looking into it. The second approach of inlining the SVGs into the markup doesn't suffer from that problem but unfortunately makes it a bit trickier for us to scale the illustrations correctly.

I'm hoping we can get back to investigating these approaches and finding a way to accurately colorize our illustrations but for now I've had to settle for the quick-n-dirty approach of regular CSS filters.

### TODO

These are the general TODOs that I've got on my list. I imagine we'll land this PR without solving most of them and instead address them in subsequent PRs but that's TBD.

- [x] I've yet to test this and adjust for Windows, that'll be top priority.
  - [ ] App menus need a little work
- [x] PR badge looks awful in dark mode when branch dropdown is open
- [ ] We've decided to not apply or support at dark theme in the welcome flow. Users should really only see that flow once at the initial startup before they have the option to select a theme. Not styling the welcome flow saves us a ton of time since it's illustration heavy. This escape hatch needs to be implemented. - #4919 
- [x] The Co-Authors toggle button hover states needs to be pulled out into variables and implemented in the dark theme - https://github.com/desktop/desktop/pull/4901
- [x] The Avatar stack still uses hard-coded colors, they just happen to kinda-sorta fit in the dark theme as well - We won't deal with that now, avatars can have any range of colors anyway.
- [x] The toast-notifications (full-screen popup) and zoom-info popups needs to be customized for the dark theme - https://github.com/desktop/desktop/pull/4915
 - [ ] Theme selection in preferences - #4919 
- [x] DialogErrors needs some dark theme love - #4900
- [ ] Analytics
- [ ] Dialog backdrops can't be customized in the current approach, we'll probably live with that for a while.
- [x] Overflow shadow visible in commit description even when it shouldn't be - #4898 
  - *Note: This turned out to have always been the case but we're addressing it as part of this release anyway as it's more noticeable in the dark theme*

### CODE REVIEW

File: app/styles/_variables.scss [hunk 1]
```diff
@@ -345,7 +345,6 @@ $overlay-background-color: rgba(0, 0, 0, 0.4);
 
   /** Dialog */
   --dialog-warning-color: $yellow-600;
-  --dialog-warning-text-color: $yellow-800;
```
Comment (niik) [inline app/styles/_variables.scss]: This variable was not used anywhere so I dropped it. It's not necessary for this PR but it's such a minor change that I didn't quite have the energy to pull it out into a separate PR.

**Comment by niik** [comment]
This is ready for review. Note that I don't expect the reviewer to manually validate all of these color choices that I've made. We'll address those in an ongoing fashion. The review here should focus on whether enabling the dark theme manually and reloading works and whether the app in dark mode looks decent without anything major standing out.

There's a few tasks in the PR body that's not done yet and that's also fine. I'll tackle those in follow-up PRs.

**Comment by nerdneha** [comment]
here's the weird background:
screen shot 2018-06-12 at 9 45 28 am

**Comment by niik** [comment]
> here's the weird background:

Yeah, that's the salmon pink brutalism color, courtesy of @joshaber in this particular instance. It's a class that's defined in our `globals.scss` 

screen shot 2018-06-12 at 18 53 41

It's only used as a means to ensure that @github/design never let's a feature through that hasn't been fully designed yet 😛 It's so ugly that we can't miss it. Additionally, and more importantly, it's only ever used for things hidden under a feature flag such as `__DEV__`

It serves its purpose as well in the dark theme as it does in the light theme.

**Comment by niik** [comment]
Thumb up all around! Gonna get this merged so we can get #4919 reviewed and merged

**Comment by nehayward** [comment]
Is there anyway to automate this from the CLI? I'm looking to do some further testing with it?

**Comment by shiftkey** [comment]
@nehayward not currently
